### PR TITLE
pytest: make loadgroup the default

### PR DESCRIPTION
**Pull request by sshane**
`loadgroup` falls back to `load` which was the default before https://github.com/commaai/openpilot/pull/30441 was merged. `loadscope` was only needed for test_models because `setUpClass` was so slow (downloading and mainly decompressing log), it was faster to not run that set up function each time in a new process for some process number configurations.

`loadscope` groups all the test cases from each class into a separate process, rather than splitting each test case up into new processes which will be slower for the majority of test classes (test_car_interfaces.py for example).

### CODE REVIEW

File: .github/workflows/selfdrive_tests.yaml [hunk 1]
```diff
@@ -73,7 +73,7 @@ jobs:
       matrix:
         arch: ${{ fromJson(
            ((github.repository == 'commaai/openpilot') &&
-              ((github.event_name != 'pull_request') || 
+              ((github.event_name != 'pull_request') ||
```
Comment (sshane) [inline .github/workflows/selfdrive_tests.yaml:76]:
```suggestion
              ((github.event_name != 'pull_request') || 
```

File: .github/workflows/selfdrive_tests.yaml [hunk 2]
```diff
@@ -226,7 +226,7 @@ jobs:
   regen:
     name: regen
     runs-on: 'ubuntu-20.04'
-    steps: 
+    steps:
```
Comment (sshane) [inline .github/workflows/selfdrive_tests.yaml:229]:
```suggestion
    steps: 
```

File: .github/workflows/selfdrive_tests.yaml [hunk 3]
```diff
@@ -189,7 +189,7 @@ jobs:
   process_replay:
     name: process replay
     runs-on: ${{ ((github.repository == 'commaai/openpilot') &&
-                   ((github.event_name != 'pull_request') || 
+                   ((github.event_name != 'pull_request') ||
```
Comment (sshane) [inline .github/workflows/selfdrive_tests.yaml:192]:
```suggestion
                   ((github.event_name != 'pull_request') || 
```

**Comment by jnewb1** [comment]
I think we are going to want this elsewhere too, what do you think of this decorator?

```
def xdist_group_property(property):
    """Mark a class based on a property of that class"""
    def decorator(cls):
        return pytest.mark.xdist_group(name=getattr(cls, property))(cls)
    return decorator
   ```
   
 ```
 @xdist_group_property('car_model')
class TestCarModel(TestCarModelBase):
```

**Comment by jnewb1** [comment]
there will also be quite a few unittests that are going to want to be grouped

**Comment by sshane** [comment]
That only marks TestCarModel (the base class that is skipped) in this case since parameterized creates the classes dynamically.

**Comment by jnewb1** [comment]
> That only marks TestCarModel (the base class that is skipped) in this case since parameterized creates the classes dynamically.

Hmm, does this currently work? I don't think so, because setup is only run on the workers so I think it would be too late to distribute them

**Comment by sshane** [comment]
Yeah, we'd need some sort of callback ability, but sadly `class_name_func` just calls the base class.

**Comment by jnewb1** [comment]
perhaps its better to handle this within conftest, which will have the class already expanded we can group it properly https://github.com/commaai/openpilot/pull/30466

**Comment by sshane** [comment]
If it works as expected, great

**Pull request closed without merging**